core/os/input_event.cpp

```diff
@@ -62,10 +62,16 @@ bool InputEvent::is_action(const StringName &p_action) const {
 
 bool InputEvent::is_action_pressed(const StringName &p_action) const {
 
-	return false; // InputMap::get_singleton()->event_is_action(Ref<InputEvent>(this),p_action);
+	if (is_action(p_action)) {
+		return is_pressed();
```

Comment:
Actually, just thought about it, I guess you might want to add `&& !is_echo()` here, but I'm not completely sure about that.